### PR TITLE
[openwrt] fix leaderpartitionid

### PR DESCRIPTION
**Pull request by simonlingoogle**

### CODE REVIEW

File: src/openwrt/ubus/otubus.hpp [hunk 1]
```diff
@@ -424,11 +424,11 @@ class UbusServer
...
-                                            const char *              aMethod,
-                                            struct blob_attr *        aMsg);
+    static int UbusPreferredPartitionIdHandler(struct ubus_context *     aContext,
+                                               struct ubus_object *      aObj,
+                                               struct ubus_request_data *aRequest,
+                                               const char *              aMethod,
+                                               struct blob_attr *        aMsg);
 
     /**
      * This method handle ubus set learderpartitionid function request.
```
Comment (zhanglongxia) [inline src/openwrt/ubus/otubus.hpp:434]: nit: preferredpartitionid

**Review by zhanglongxia — APPROVED** [review]
LGTM

File: src/openwrt/ubus/otubus.hpp (hunk 1 above)
Comment (simonlingoogle) [reply src/openwrt/ubus/otubus.hpp:434]: fixed

**Review by bukepo — APPROVED** [review]

**Review by librasungirl — APPROVED** [review]

**Comment by jwhui** [comment]
@simonlingoogle , the openwrt-check on Travis is currently failing.

**Comment by simonlingoogle** [comment]
> @simonlingoogle , the openwrt-check on Travis is currently failing.

I am aware of that. 

Discussed with @bukepo, we agreed that we can change `preferredpartitionid` to `partitionid` because 1) `otThreadGetPreferredLeaderPartitionId` is only available for reference device, and 2) displaying `preferred partition id` makes less sense than the actual `partition id`.

File: src/openwrt/view/admin_thread/thread_setting.htm [hunk 2]
```diff
@@ -160,23 +160,13 @@ <h3><%:Network Configuration%></h3>
...
-			<input type="text" name="leaderpartitionid" value="<%=threadget("leaderpartitionid").Leaderpartitionid%>" style="width:50%;"/>
-			</div>
-			<div style="margin-left:30%;margin-top:1%;">
-				<span><img src="<%=resource .. "/cbi/help.gif"%>"/><%:leaderpartitionid must be a number%></span>
-			</div>
-			<% else %>
-			<label class="cbi-value-title" style="margin-right:5%;">LeaderPartitionId</label>
-			<div class="cbi-value-title">
-			<input type="text" name="leaderpartitionid" value="<%=threadget("leaderpartitionid").Leaderpartitionid%>" readonly="readonly" style="width:50%;"/>
+			<input type="text" name="partitionid" value="<%=threadget("partitionid").Partitionid%>" readonly="readonly" style="width:50%;"/>
```
Comment (bukepo) [inline src/openwrt/view/admin_thread/thread_setting.htm:165]:
```suggestion
			<%=threadget("partitionid").Partitionid%>
```

File: src/openwrt/view/admin_thread/thread_setting.htm [hunk 3]
```diff
@@ -393,15 +383,15 @@ <h3><%:Interface Configuration%></h3>
...
 			}
 
-			var leaderpartitionid = document.forms["settingForm"]["leaderpartitionid"].value;
-			if (isNaN(leaderpartitionid)) {
-				alert("leaderpartitionid must be a number");
+			var partitionid = document.forms["settingForm"]["partitionid"].value;
+			if (isNaN(partitionid)) {
+				alert("partitionid must be a number");
 				return false;
 			}
```
Comment (bukepo) [inline src/openwrt/view/admin_thread/thread_setting.htm:390]: I don't think this is necessary, we can remove 386-388 and 392-394
```suggestion
```

File: src/openwrt/view/admin_thread/thread_setting.htm [hunk 4]
```diff
@@ -393,15 +383,15 @@ <h3><%:Interface Configuration%></h3>
...
```
Comment (bukepo) [inline src/openwrt/view/admin_thread/thread_setting.htm:396]:
```suggestion
```

File: src/openwrt/view/admin_thread/thread_setting.htm (hunk 2 above)
Comment (simonlingoogle) [reply src/openwrt/view/admin_thread/thread_setting.htm:165]: should be fine to use a readonly text?

Comment (bukepo) [reply src/openwrt/view/admin_thread/thread_setting.htm:165]: it will still be submitted even it's readonly.

Comment (simonlingoogle) [reply src/openwrt/view/admin_thread/thread_setting.htm:165]: Should be fine if we don't use `partitionid` arguments?

Comment (bukepo) [reply src/openwrt/view/admin_thread/thread_setting.htm:165]: but why submit something not to be used?

**Comment by simonlingoogle** [comment]
WIP :Unable to pass LGTM C++ check and the error log seems not provide any meaningful information.

**Comment by jwhui** [comment]
> Should we disable LGTM.com check for now to unblock PRs? @jwhui

A fix has been submitted https://github.com/github/codeql/issues/4809#issuecomment-743142125. I kicked LGTM.com to see if it will pass now.

**Review by jwhui — APPROVED** [review]

**Pull request merged**